blog(embertimes107): RFC intimate API (#193)
ember-learn/ember-blog: source/2019-07-19-the-ember-times-issue-107.md

---

## [RFC: Major Version Release for All Deprecation Removal 📋](https://github.com/emberjs/rfcs/pull/512)

An RFC was proposed by [@kerrick](https://github.com/Kerrick) to have a major version release for all deprecation removals. The main issue revolves around the [Intimate API](https://twitter.com/wycats/status/918644693759488005) policy which can catch some developers off guard when upgrading Ember versions.

> This is because not all private APIs would warrant a deprecation warning (and therefore a major version change). The RFC proposes to have a major version release for all deprecation removals.

Reviewer: I'm not that into Ember, but I cannot really follow this sentence. Is there a way to make it better understandable?

Author: Fair point, ive added some more to try to break it down

> Since certain private APIs are considered "Intimate API" due to the wide range of addons and consumers who use it, despsite it being a private method we sometimes have issues when deprecating these private APIs. This is because not all private APIs would warrant a deprecation warning (and therefore a major version change). The RFC proposes to have a major version release for all deprecation removals.

Reviewer 2: I think you can rephrase the first sentence for clarity.
```suggestion
Certain private APIs are considered "Intimate API" because there are addons and consumers who use them despite being private. We sometimes have issues when deprecating these private APIs. This is because not all private APIs would warrant a deprecation warning (and therefore a major version change). The RFC proposes to have a major version release for all deprecation removals.
```
